core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -1288,16 +1297,6 @@ object Pull extends PullLowPriority {
     go(initScope, None, initFk, new OuterRun(init), stream)
   }
 
-  private[fs2] def flatMapOutput[F[_], F2[x] >: F[x], O, O2](
-      p: Pull[F, O, Unit],
-      f: O => Pull[F2, O2, Unit]
-  ): Pull[F2, O2, Unit] =
-    p match {
-      case a: AlgEffect[F, Unit] => a
-      case r: Terminal[_]        => r
-      case _                     => FlatMapOutput(p, f)
-    }
```

Comment:
Instead of adding a MiMa filter, can we just make this deprecated and delegate it to the new implementation?

Reply:
Hmmmm if it is a private[fs2] method, how could a binary incompatibility happen?

Reply:
It's because of https://github.com/lampepfl/dotty/issues/10842.

It would be safe to add a filter, but every filter we add risks accidentally masking a real issue. So my personal preference is to avoid filters even for private stuff if there's an easy fix.

Reply:
Not ideal, but so be it. Added back.